app/presenters/person_presenter.rb

```diff
@@ -34,14 +31,12 @@ def full_hash_with_profile
   end
 
   def as_json(_options={})
-    attrs = full_hash_with_avatar
-
-    if own_profile? || person_is_following_current_user
-      attrs.merge!(
-        location: @presentable.location,
-        birthday: @presentable.formatted_birthday,
-        bio:      @presentable.bio
-      )
+    attrs = full_hash
+
+    if attrs[:show_profile_info]
+      attrs.merge!(ProfilePresenter.new(profile).private_hash) {|_, v1, _| v1 }
+    else
+      attrs.merge!(ProfilePresenter.new(profile).public_hash) {|_, v1, _| v1 }
     end
```

Comment:
No, I was wrong. I receive messages "Unexpected modal dialog (text: A script on this page may be busy, or it may have stopped responding. You can stop the script now, or you can continue to see if the script will complete." with both versions of the as_json function. Dunno how to deal with that. Probably my test machine is too weak.

However, I found out, that the as_json function doesn't get called during run of the cucumber testsuite. So its changes can't probably break anything at all.